src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -607,19 +618,70 @@ def mask_type(self, value: Union[str, List[int], PruningMaskCreator]):
         if not isinstance(value, PruningMaskCreator):
             self._mask_creator = load_mask_creator(value)
 
+    @ModifierProp()
     def global_sparsity(self) -> bool:
         """
         :return: True if global pruning is enabled, False otherwise
         """
         return self._global_sparsity
 
+    @ModifierProp()
+    def score_type(self) -> PruningScoreTypes:
+        """
+        :return: the the scoring method used for pruning
+        """
+        return self._score_type
+
     @ModifierProp(serializable=False)
     def applied_sparsity(self) -> float:
         """
         :return: the currently applied sparsity level to the contained params
         """
         return self._applied_sparsity
 
+    def optimizer_pre_step(
+        self, module: Module, optimizer: Optimizer, epoch: float, steps_per_epoch: int
+    ):
+        """
+        Update mask movement scores with gradients right before optimizer step is
+        applied. Called here in case gradients are changed between the backwards
+        pass and step such as in grad norm clipping
+        :param module: module to modify
+        :param optimizer: optimizer to modify
+        :param epoch: current epoch and progress within the current epoch
+        :param steps_per_epoch: number of steps taken within each epoch
+            (calculate batch number using this and epoch)
+        """
+        super().optimizer_pre_step(module, optimizer, epoch, steps_per_epoch)
+
+        if self._module_masks.score_type == PruningScoreTypes.MOVEMENT:
+            if not self._movement_scores_updated:
+                self._module_masks.update_movement_scores()
+            self._movement_scores_updated = False
+
+    def optimizer_post_step(
+        self, module: Module, optimizer: Optimizer, epoch: float, steps_per_epoch: int
+    ):
+        """
+        Reapply the mask after the optimizer step in case the optimizer
+        has momentum that may have moved weights from 0. Not applied for
+        movement pruning to allow weight reintroduction
+
+        :param module: module to modify
+        :param optimizer: optimizer to modify
+        :param epoch: current epoch and progress within the current epoch
+        :param steps_per_epoch: number of steps taken within each epoch
+            (calculate batch number using this and epoch)
+        """
+        super(_PruningParamsModifier, self).optimizer_post_step(
+            module, optimizer, epoch, steps_per_epoch
+        )
+
+        if self._module_masks.score_type != PruningScoreTypes.MOVEMENT:
```

Comment:
Same as above, can we make this generic and override in mvp mod?

Reply:
Let's push comments like these off to a refactor later. Priority is to get this landed ASAP for release.

Reply:
updated to use `allow_reintroducion` to make this more generic